code/ship/ship.cpp

```diff
@@ -3291,10 +3302,12 @@ int parse_ship_values(ship_info* sip, const bool is_template, const bool first_t
 		Error(LOCATION, "%s '%s' has too many primary banks (%d).  Maximum for ships is currently %d.\n", info_type_name, sip->name, sip->num_primary_banks, MAX_SHIP_PRIMARY_BANKS);
 	}
 
+	memset(sip->allowed_weapons, 0, sizeof(int) * MAX_WEAPON_TYPES);
```

Comment:
Changing every instance is probably not a good idea because it's likely to introduce bugs but the better version of determining the size of the array should be used whenever possible.